python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -626,6 +627,24 @@ def __set__(self, obj, _val):
         GangaObject.__setattr__(raw_obj, attr_name, final_val)
 
 
+def proxy_wrap(f):
+    # type: (Callable) -> Callable
+    """
+    A decorator to strip the proxy from all incoming arguments
+    (including ``self`` if it's a method) and add one to the return
+    value.
+    """
+
+    @functools.wraps(f)
+    def proxy_wrapped(*args, **kwargs):
+        s_args = [stripProxy(a) for a in args]
```

Comment:
Indeed, things with non-automatic proxies (like the registries etc.) are a potential leak in the proxy layer but they should be fixed by making them conform to the `_impl`/`_proxyObject` interface so that `assert a is addProxy(stripProxy(a))` always works.

The registries aren't usually passed through the proxy layer though so they're unlikely to be a problem in the short-term.